use-shopping-cart/src/reducers.js

```diff
@@ -48,25 +48,59 @@ export const cartValuesInitialState = {
   totalPrice: 0,
   cartCount: 0
 }
-function Entry(productData, quantity, currency, language) {
+
+function Entry(
+  product,
+  quantity,
+  currency,
+  language,
+  price_metadata,
+  product_metadata
+) {
   const getProductId = () => {
-    if (productData.id) {
-      return productData.id
+    if (product.id) {
+      return product.id
     }
-    if (productData.price_id) {
-      return productData.price_id
+    if (product.price_id) {
+      return product.price_id
     }
-    if (productData.sku_id) {
-      return productData.sku_id
+    if (product.sku_id) {
+      return product.sku_id
     }
-    if (productData.sku) {
-      return productData.sku
+    if (product.sku) {
+      return product.sku
     }
     return uuidv4()
   }
 
+  if (!product.price_data && price_metadata) {
+    product.price_data = {}
```

Comment:
This line is a bit redundant since in the next line we're overwriting it with a new object with the `price_metadata` in it.

Reply:
I think what I'm trying to account for is if the product that is already in the cart, doesn't have price_data in it, then add the price_data object first.

The second if statement straight up spreads price_metadata to product if it already exists. Is this still redundant? Am I misunderstanding something here?

Reply:
You're understanding it right but line 77 is overwritten by line 79.

```js
if (!product.price_data && price_metadata) {
  // product.price_data is undefined
  
  product.price_data = {}
  // product.price_data is now {} (an empty object)

  product.price_data = {
    ...price_metadata
  }
  // product.price_data is now a copy of price_metadata (previously line no longer matters bc of this).
}
```

If I was to take out line 77, line 79 would still do the same thing and produce the same outcome.